src/services/commands.js

```diff
@@ -0,0 +1,153 @@
+Controller.open(function(_) {
+
+  var COMMAND_CONFIGURATION_TYPE_MAP = {
+    symbol: 'vanillaSymbol',
+    variable: 'variable',
+    nonSymbola: 'nonSymbolaSymbol',
+    variable: 'binary'
+  }
+
+  _.initializeLatexGrammar = function() {
+    this.latexCmds = Object.assign({}, LatexCmds);
+    this.charCmds = Object.assign({}, CharCmds);
+    this.textCommands = {}
+
+    this.cursor.options.autoCommands = {}
+    
+    // Initialize the grammar processors for various symbols
+    // by loading up the default configuration 
+    // This is WIP. Eventually all files symbol definitions
+    // should be transferred over to the defaultSymbolDefinitions file
+    this.extendLatexGrammar(GLOBALLY_DISABLED_INPUT, 'symbol');
+    // this.extendLatexGrammar(VANILLA_SYMBOLS, "symbol");
+    // this.extendLatexGrammar(NON_SYMBOLA_SYMBOLS, "nonSymbola");
+    // this.extendLatexGrammar(GREEK_SYMBOLS, "variable");
+    // this.extendLatexGrammar(BINARY_SYMBOLS, "variable");
+
+    // Process injected commands into autocommands 
+    var options = this.cursor.options;
+    var commands = options.commands || []
+    this.extendLatexGrammar(commands)  
+  }
+
+  _.attachKeyboardListener = function (grammarDefinition) {
+    if (grammarDefinition.keystroke) {
+      // this is still a bit hacky and we may get performance issues in the future
+      // but for now it works. 
+    
+      this.registerKeystrokeHandler(function(pattern,event) {
+       
+        if (pattern === grammarDefinition.keystroke) {
+          event.preventDefault();
+          this.API.cmd(grammarDefinition.name);
+        }
+      });
+      // we want to make sure that keystroke inputs do not generate any markup 
+      // this means overriding the 
+    }
+    
```

Comment:
**nit** Okay so maybe I have many nits about blank lines, but this is ridiculous! :wink: